perllib/FixMyStreet/App/Form/Waste/Garden/Modify.pm

```diff
@@ -4,25 +4,27 @@ use utf8;
 use HTML::FormHandler::Moose;
 extends 'FixMyStreet::App::Form::Waste';
 
-has_field is_staff => ( type => 'Hidden' );
-
 has_page intro => (
     title => 'Modify your green garden waste subscription',
     template => 'waste/garden/modify_pick.html',
     fields => ['task', 'continue'],
-    update_field_list => sub {
-        my $form = shift;
-        return {
-            is_staff => { default => $form->{c}->stash->{is_staff} }
-        };
-    },
-    next => sub { return $_[0]->{is_staff} ? 'alter_staff' : 'alter'; },
+    next => 'alter',
 );
 
-my %alter_fields = (
+has_page alter_all => ( # Dummy so that fields are hidden if not staff below
+    fields => ['current_bins', 'bins_wanted', 'name', 'phone', 'email', 'continue_review'],
+);
```

Comment:
Just to check, this is because the form engine shows fields that aren't on a specific page on all pages so we need to have a page that shows optional fields so they aren't on all pages?

Reply:
Yes, exactly (our code in Wizard.pm marks all fields in pages that are not in `current_page->all_fields` as inactive, to be precise). I couldn't think of an easier solution. I guess it could be an attribute on the page itself and then alter that Wizard.pm code in build_active to read that to make sure they're marked inactive too, that might be bit cleaner?

Reply:
Got rid of these silly extra pages and put fields back (with full list) and added a runtime `field_ignore_list` which returns list of fields to not include this time. Simple.pm slightly more complicated code, but all other boilerplate simpler: 7b235e0039187d53b80dadc70c6e3e846c61db95